### ISSUE TITLE
Errors when compiling mseed_simple.c

### ISSUE THREAD

**Issue by Sparrow0hawk** [emolch/kiwi#21]
I'm trying to build kiwi on a CentOS Linux release 7.6 cluster machine. 

I'm using HDF5 (1.8.21) and FFTW (3.3.8), ifort 19.0.4. I've installed libmseed locally and have exported the `include` and `lib` dirs into `$CPATH` and `$LIBRARY_PATHS`. However I get the following errors when running `make` with my adapted `Makefile.local` in which I'm specifying the non-typical locations of HDF5 and FFTW.

```
 Entering directory `/home/home01/medacola/tmp_rocko/kiwi/mseed'
gcc -c -std=c99 -Wall  mseed_simple.c
mseed_simple.c: In function ‘writemseed’:
mseed_simple.c:15:5: error: unknown type name ‘MSTrace’
     MSTrace     *trace;
     ^
mseed_simple.c:20:5: error: unknown type name ‘hptime_t’
     hptime_t    hptime;
     ^
mseed_simple.c:28:5: warning: implicit declaration of function ‘mst_init’ [-Wimplicit-function-declaration]
     trace = mst_init (NULL);
     ^
mseed_simple.c:28:11: warning: assignment makes pointer from integer without a cast [enabled by default]
     trace = mst_init (NULL);
           ^
mseed_simple.c:30:19: error: request for member ‘network’ in something not a structure or union
     strncpy( trace->network, network, 2 );
                   ^
mseed_simple.c:31:19: error: request for member ‘station’ in something not a structure or union
     strncpy( trace->station, station, 4 );
                   ^
mseed_simple.c:32:19: error: request for member ‘location’ in something not a structure or union
     strncpy( trace->location, location, 2 );
                   ^
mseed_simple.c:33:19: error: request for member ‘channel’ in something not a structure or union
     strncpy( trace->channel, channel, 3 );
                   ^
mseed_simple.c:34:10: error: request for member ‘network’ in something not a structure or union
     trace->network[3]  = '\0';
          ^
mseed_simple.c:35:10: error: request for member ‘station’ in something not a structure or union
     trace->station[5]  = '\0';
          ^
mseed_simple.c:36:10: error: request for member ‘location’ in something not a structure or union
     trace->location[3] = '\0';
          ^
mseed_simple.c:37:10: error: request for member ‘channel’ in something not a structure or union
     trace->channel[4]  = '\0';
          ^
mseed_simple.c:45:15: error: ‘hptime_t’ undeclared (first use in this function)
     hptime = (hptime_t)(sec_em5 * HPTMODULUS);
               ^
mseed_simple.c:45:15: note: each undeclared identifier is reported only once for each function it appears in
mseed_simple.c:45:35: error: ‘HPTMODULUS’ undeclared (first use in this function)
     hptime = (hptime_t)(sec_em5 * HPTMODULUS);
                                   ^
mseed_simple.c:47:10: error: request for member ‘starttime’ in something not a structure or union
     trace->starttime = hptime;
          ^
mseed_simple.c:48:10: error: request for member ‘samprate’ in something not a structure or union
     trace->samprate = 1./deltat;
          ^
mseed_simple.c:49:10: error: request for member ‘sampletype’ in something not a structure or union
     trace->sampletype = 'f';
          ^
mseed_simple.c:55:10: error: request for member ‘datasamples’ in something not a structure or union
     trace->datasamples = seismogram;
          ^
mseed_simple.c:56:10: error: request for member ‘numsamples’ in something not a structure or union
     trace->numsamples = length;
          ^
mseed_simple.c:57:10: error: request for member ‘samplecnt’ in something not a structure or union
     trace->samplecnt = length;
          ^
mseed_simple.c:58:10: error: request for member ‘sampletype’ in something not a structure or union
     trace->sampletype = 'f';
          ^
mseed_simple.c:59:5: warning: implicit declaration of function ‘mst_srcname’ [-Wimplicit-function-declaration]
     mst_srcname (trace, srcname, 0);
     ^
mseed_simple.c:61:5: warning: implicit declaration of function ‘mst_pack’ [-Wimplicit-function-declaration]
     precords = mst_pack (trace, &record_handler, outfile, 4096, DE_FLOAT32,
     ^
mseed_simple.c:64:5: warning: implicit declaration of function ‘mst_free’ [-Wimplicit-function-declaration]
     mst_free( &trace );
     ^
mseed_simple.c:17:27: warning: variable ‘precords’ set but not used [-Wunused-but-set-variable]
     int         psamples, precords;
                           ^
mseed_simple.c: In function ‘readmseed’:
mseed_simple.c:72:5: error: unknown type name ‘MSTraceGroup’
     MSTraceGroup *mstg = NULL;
     ^
mseed_simple.c:73:5: error: unknown type name ‘MSTrace’
     MSTrace      *trace;
     ^
mseed_simple.c:84:5: error: unknown type name ‘hptime_t’
     hptime_t hptime;
     ^
mseed_simple.c:88:5: warning: implicit declaration of function ‘ms_readtraces’ [-Wimplicit-function-declaration]
     retcode = ms_readtraces (&mstg, filename, 0, -1.0, -1.0, 0, 1, 1, 0);
     ^
mseed_simple.c:97:17: error: request for member ‘traces’ in something not a structure or union
     trace = mstg->traces;
                 ^
mseed_simple.c:99:14: error: request for member ‘datasamples’ in something not a structure or union
     if (trace->datasamples == NULL) {
              ^
mseed_simple.c:104:48: error: request for member ‘numsamples’ in something not a structure or union
     data = (float*) calloc(sizeof(float), trace->numsamples);
                                                ^
mseed_simple.c:106:14: error: request for member ‘sampletype’ in something not a structure or union
     if (trace->sampletype == 'i') {
              ^
mseed_simple.c:107:33: error: request for member ‘datasamples’ in something not a structure or union
         source_int = (int*)trace->datasamples;
                                 ^
mseed_simple.c:108:26: error: request for member ‘numsamples’ in something not a structure or union
         for (i=0; i<trace->numsamples; i++) {
                          ^
mseed_simple.c:112:14: error: request for member ‘sampletype’ in something not a structure or union
     if (trace->sampletype == 'a') {
              ^
mseed_simple.c:113:35: error: request for member ‘datasamples’ in something not a structure or union
         source_char = (char*)trace->datasamples;
                                   ^
mseed_simple.c:114:26: error: request for member ‘numsamples’ in something not a structure or union
         for (i=0; i<trace->numsamples; i++) {
                          ^
mseed_simple.c:118:14: error: request for member ‘sampletype’ in something not a structure or union
     if (trace->sampletype == 'f') {
              ^
mseed_simple.c:119:37: error: request for member ‘datasamples’ in something not a structure or union
         source_float = (float*)trace->datasamples;
                                     ^
mseed_simple.c:120:26: error: request for member ‘numsamples’ in something not a structure or union
         for (i=0; i<trace->numsamples; i++) {
                          ^
mseed_simple.c:124:14: error: request for member ‘sampletype’ in something not a structure or union
     if (trace->sampletype == 'd') {
              ^
mseed_simple.c:125:39: error: request for member ‘datasamples’ in something not a structure or union
         source_double = (double*)trace->datasamples;
                                       ^
mseed_simple.c:126:26: error: request for member ‘numsamples’ in something not a structure or union
         for (i=0; i<trace->numsamples; i++) {
                          ^
mseed_simple.c:131:19: error: request for member ‘starttime’ in something not a structure or union
     hptime = trace->starttime ;
                   ^
mseed_simple.c:134:39: error: ‘HPTMODULUS’ undeclared (first use in this function)
     dtime = (double)hptime / ((double)HPTMODULUS);
                                       ^
mseed_simple.c:136:20: error: request for member ‘numsamples’ in something not a structure or union
     *length = trace->numsamples;
                    ^
mseed_simple.c:138:23: error: request for member ‘samprate’ in something not a structure or union
     *deltat = 1./trace->samprate;
                       ^
mseed_simple.c:141:5: warning: implicit declaration of function ‘mst_freegroup’ [-Wimplicit-function-declaration]
     mst_freegroup (&mstg);
     ^
make[1]: *** [mseed_simple.o] Error 1
make[1]: Leaving directory `/home/home01/medacola/tmp_rocko/kiwi/mseed'
make: *** [.mseedsimple] Error 2

```

Any advice would be much appreciated.

**Comment by Sparrow0hawk** [comment]
Closing as this is a duplicate of https://github.com/emolch/kiwi/issues/19

It seems kiwi requires libmseed version <= 2.19.6
